{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Fetching synonyms from CZI and SoftwareKG dataset \n",
    "\n",
    "Considering that CZI has synonyms fetched from most used software websites (PyPi, CRAN, Bioconductor and SciCrunch) this notebook will for every software in the benchmark query the datasets and try to retrieve synonyms. \n",
    "\n",
    "First we save all unique software names (lowered) into a dictionary, the using a function get_synonyms_CIZ, we find all synonyms that have software_mention (lowered) the same as the key in the dictionary. Dictionary is used in order to speed up the process.\n",
    "\n",
    "Next, we query SoftwareKG graphs for synonyms (lowered). Again, the dictionary is used to speed up the process by querieng only once for the same name. This is done in get_synonyms_from_SoftwareKG function.\n",
    "\n",
    "CZI dataset was downloaded from https://datadryad.org/dataset/doi:10.5061/dryad.6wwpzgn2c#methods (disambiguated file).\n",
    "SoftwareKG SPARQL point: https://data.gesis.org/somesci/sparql"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "#Read files\n",
    "df = pd.read_csv(\"../CZI/synonyms_matrix.csv\")\n",
    "benchmark_df = pd.read_csv(\"../temp/v3.1/updated_with_metadata_file.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: sparqlwrapper in c:\\users\\jelena\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (2.0.0)\n",
      "Requirement already satisfied: rdflib>=6.1.1 in c:\\users\\jelena\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from sparqlwrapper) (7.0.0)\n",
      "Requirement already satisfied: isodate<0.7.0,>=0.6.0 in c:\\users\\jelena\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from rdflib>=6.1.1->sparqlwrapper) (0.6.1)\n",
      "Requirement already satisfied: pyparsing<4,>=2.1.0 in c:\\users\\jelena\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from rdflib>=6.1.1->sparqlwrapper) (3.1.4)\n",
      "Requirement already satisfied: six in c:\\users\\jelena\\appdata\\local\\programs\\python\\python312\\lib\\site-packages (from isodate<0.7.0,>=0.6.0->rdflib>=6.1.1->sparqlwrapper) (1.16.0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "[notice] A new release of pip is available: 25.0.1 -> 25.1.1\n",
      "[notice] To update, run: python.exe -m pip install --upgrade pip\n"
     ]
    }
   ],
   "source": [
    "%pip install sparqlwrapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "#check for null values and drop them\n",
    "#turning all names into strings\n",
    "print(df[\"software_mention\"].isnull().sum())\n",
    "df = df.dropna(subset=[\"software_mention\"])\n",
    "print(df[\"software_mention\"].isnull().sum())\n",
    "df['software_mention'] = df['software_mention'].astype(str)  # Convert all values to strings\n",
    "benchmark_df['name']=benchmark_df['name'].astype(str)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#making a dictionary for the benchmark\n",
    "benchmark_dictonary = {name.lower(): set() for name in benchmark_df[\"name\"].unique()}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Function that retirieves synonyms for each software mention\n",
    "def get_synonyms_from_CZI(df, dictionary):\n",
    "    for key in dictionary.keys():\n",
    "        # Find matching rows in synonyms_df where the software mention matches the dictionary key\n",
    "        matches = df[df[\"software_mention\"].str.lower() == key][\"synonym\"].tolist()\n",
    "        # Store synonyms as a list\n",
    "        dictionary[key].update(matches)\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "from SPARQLWrapper import SPARQLWrapper, JSON\n",
    "\n",
    "\n",
    "def get_synonyms_from_SoftwareKG(dictionary):\n",
    "    # Define the SPARQL endpoint\n",
    "    sparql = SPARQLWrapper(\"https://data.gesis.org/somesci/sparql\")\n",
    "    # Execute the query\n",
    "    for key in dictionary.keys():\n",
    "        query = f\"\"\"\n",
    "    PREFIX nif: <http://persistence.uni-leipzig.org/nlp2rdf/ontologies/nif-core#>\n",
    "PREFIX sms: <http://data.gesis.org/somesci/>\n",
    "PREFIX its: <http://www.w3.org/2005/11/its/rdf#>\n",
    "\n",
    "SELECT DISTINCT ?synonym\n",
    "WHERE {{\n",
    "    # Find the software entity associated with the given spelling\n",
    "    ?sw_phrase a nif:Phrase ;\n",
    "               its:taClassRef [ rdfs:subClassOf sms:Software ] ;\n",
    "               its:taIdentRef ?sw_identity ;\n",
    "               nif:anchorOf \"{key}\" .  # Replace \"Excel\" with the desired software name\n",
    "\n",
    "    # Retrieve other spellings linked to the same software identity\n",
    "    ?other_phrase its:taIdentRef ?sw_identity ;\n",
    "                  nif:anchorOf ?synonym .\n",
    "    \n",
    "    FILTER (?synonym != \"{key}\")  # Exclude the original input spelling from results\n",
    "}}\n",
    "ORDER BY ?synonym\n",
    "    \"\"\"\n",
    "        try:\n",
    "            # Set query and return format\n",
    "            sparql.setQuery(query)\n",
    "            sparql.setReturnFormat(JSON)\n",
    "            results = sparql.query().convert()\n",
    "\n",
    "            # Process results\n",
    "            for result in results[\"results\"][\"bindings\"]:\n",
    "                synonym = result.get(\"synonym\", {}).get(\"value\")\n",
    "                if synonym:\n",
    "                    dictionary[key].add(synonym)\n",
    "\n",
    "        except Exception as e:\n",
    "            print(f\"Error retrieving synonyms for {key}: {e}\")\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_synonyms(dictionary, CZI = 1, SoftwareKG = 1):\n",
    "    if CZI == 1:\n",
    "        get_synonyms_from_CZI(df, dictionary)\n",
    "    if SoftwareKG == 1:\n",
    "        get_synonyms_from_SoftwareKG(dictionary)\n",
    "    dictionary = {key: \", \".join(value) for key, value in dictionary.items()}\n",
    "    return dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error retrieving synonyms for (star: <urlopen error [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond>\n",
      "Error retrieving synonyms for abbyy: <urlopen error [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond>\n",
      "Error retrieving synonyms for bcea: <urlopen error [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond>\n",
      "Error retrieving synonyms for beautifulsoup: <urlopen error [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond>\n",
      "Error retrieving synonyms for coapthon: <urlopen error [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond>\n",
      "{'(star': '(star, STAR, star, (STARS, steroidogenic acute regulatory protein, mitochondrial, (STARMAPs, (STARS), (STARDIT), start domain-containing protein 1, luteinizing hormone-induced protein, stard1, (STARD, (STAR), (START)', 'abbyy': 'ABBYY OCR', 'bcea': 'R-package BCEA, R package BCEA (Bayesian Cost Effectiveness Analysis), R package BCEA, BCE', 'beautifulsoup': 'BeautifulSoup Python Package, Beautiful Soup”, beautiful-soup, BeautifulSoup9, BeautifulSoup 4 (PyPi), Beautiful, BeautifulSoup Python library, beautifulsoup4, BeatifulSoup, BeautifulSoup4, Beautiful Soup, Beautiful soup', 'coapthon': 'CoAPthon3', 'diagrammersvg': 'Diagrammer, DiagrammeR, Diagram', 'django': 'Django Web, Django Python Web framework, Djangology, django, DJANGOS, Django CMS, Django-rest, Django web, django-open, django-auth, Django Python', 'fincal': 'inCal', 'flask': 'FLASKA, Python Flask, Flask MVC, FLAS, Flask Python Framework, Flask Web, Flask Python library, Flask Python', 'gillespiessa': 'R package “GillespieSSA, R package ‘GillespieSSA’, GillespieSSA R, Gillespie2, R-package GillespieSSA, Gillespie', 'hiv.lifetables': '', 'hydenet': '', 'mg-rast': 'MG-RASTv, MG-RAST API, MG-RAST-API, mg-rast - metagenomics analysis server, MG-RAST - metagenomics analysis server, MG-RAST SEED, MG-RASTare, MG-RAST23, MG-RAST), MG RAST, Metagenomics RAST, MG-RAST18, MG-Rast M5NR, MG-RAST BLAT, MG-RAST53, MG-RAST46, MG-RAST43, The Metagenomics RAST server, the metagenomics rast server, metagenomics rast, MG-RAST12, MG-Rast), MG-RASR, MG-RASTv2, mg-rast, MG-RASTa', 'mgdrive': 'GDrive, MGDrivE, MGDrive, MGDrivE2', 'multiphen': 'R package MultiPhen (MultiPhen), Multiphor, MultiPhen-GATES, MultiPhyl, MultiPhen, MultiPhen59, Multiphen, MultiP, MultiPhen R package, R package MultiPhen, MultiPhen9, Multiphyl, Multi', 'nlme': '“nlme, R package nlme package, [51], nlme R library, nlme59, nlme51, R function nlme, nlme” R, R library nlme, R package \"nlme\", nlme library of R, nlme18, R packages nlme, nlme R, nlme-3, nlm, nlme package for, nlme), R package \\'nlme\\', nlme42, r-package nlme, R package: nlme, package nlme, nlme92, pNLME, nlme R Package, R-package ‘nlme, NLMEM, R-package “nlme”, nlme R-package, nlme package (Pinheiro ), nlme60, R package “nlme”, ≪nlme≫ R, R nlme library, r package nlme, nlme” R package, nlme-R, NLM, R package ‘nlme’, NLMETM, NLMEs, R nlme package, package “nlme, on package nlme, nlme19, nlme package for R, R-package ‘nlme’, nlme82, R package \\'nlme, nlme package of R, R language nlme package, R library “nlme”, R package ‘nlme, nlme_3, nlme’, R Package nlme, nlme’ R, R library “nlme, NLME-A, nlme R package, R « nlme » package (Linear and Nonlinear Mixed Effects Models), nlme of R, R ‘nlme’, R packge nlme, R package “nlme, R-package nlme, nlme4, R package nlme, NLME, nlme package, R nlme, nlme55, NLME R, R-package “nlme', 'panda': 'Pandas32, iPANDA, pandas®, pipeline for analyzing brain diffusion images, panda (pipeline for analyzing brain diffusion images), Pandaset, panda-tool, PANDAR, Panda3D, ANDA, PandaX-II, Pandaseq, PANDA-T1ρ, PANDA GRN, Pandanus, PANDAseq, Pandas9, PandaR, PANDASeq2, PAND, PANDAseq), PANDA (Pipeline for Analyzing braiN Diffusion imAges), PANDAT, Mpanda, PANDASEQ, iPanda, PANDA+, PANDAs, Pandas”, Pandas4, pandas0, PANDAA, PandaT, PANDDA, PANDABox, PandaEPL, Pand, PANDASeq, PandaSeq, Pandas-0, PANDANA, Pandat, Pandda, PANDA: a pipeline tool for diffusion MRI, Pandat®, panda, RPANDA, pandda, PANDA), rpanda, PANDAseq2, panda: pipeline for analyzing brain diffusion images, pandaSeq, PANDAS, pandaR, PANDAse, PandaSeq2, pandaseq, Pipeline for Analyzing braiN Diffusion imAges, pandas, Pandas©, PANDAbox, PANDA seq, Pandas, PandaS, Pandaæsim, panda: a pipeline tool for diffusion mri, Panda3d, Panda-seq, PandaX, pandaCam, PANDA: Pipeline for Analyzing braiN Diffusion imAges', 'panda (pipeline for analyzing brain diffusion images)': 'PANDA (Pipeline for Analyzing Brain Diffusion Images), PANDA (Pipeline for Analyzing, PANDA (Pipeline for Analyzing braiN Diffusion imAges), PANDA (Pipeline for Analysing Brain Diffusion Images), PANDA (Pipeline for Analyzing Brain Diffusion Images toolkit), Pipeline for Analyzing Brain Diffusion Images, Pipeline for Analyzing braiN Diffusion imAges (PANDA', 'poap': 'POA, POAPRA', 'pandas': 'Python package pandas, pandas Python library, PANDAseq42, Pandas32, PandasTools, pandas®, Pandas qcut, pipeline for analyzing brain diffusion images, panda (pipeline for analyzing brain diffusion images), Pandaset, PANDA, panda-tool, PANDAR, PANDAs PtDA, Pandaseq, PANDAseq57, Panda, PANDAseq, Pandas9, PandaR, PANDAseq28, pandas Python, PANDASeq2, PAND, PANDAseq), PANDAT, python pandas, PANDASEQ, PANDA+, ANDAS, Python pandas, PANDAs, Pandas”, Pandas4, pandas0, PANDAA, PandaT, PANDDA, Pand, PANDASeq, Pandas-0, PandaSeq, PandaSeq32, Pandat, Pandda, PANDASeq36, panda, pandda, PANDA), PANDAseq2, panda: pipeline for analyzing brain diffusion images, PANDAS, Python pandas package, pandastable, pandaR, PANDAse, python pandas scripts, PandaSeq2, pandaseq, pandas, Pandas©, PANDAs PDA, PANDAseq33, PANDAseq38, PandaS, pandas Python package, Pandas, panda: a pipeline tool for diffusion mri, PandaX', 'presenceabsence': 'R package PresenceAbsence, R package “Presence/Absence”, R package \"PresenceAbsence\", PresenceAbsence R package, R package “PresenceAbsence, R package “PresenceAbsence”, Presence, PresenceAbsence62, Presence Absence, PresenceAbsence for R, PresenceAbsence R, R package ‘PresenceAbsence’, R package \\'PresenceAbsence\\'', 'pysiology': '', 'pygments': '', 'python package sklearn': 'Python package “scikit-learn”, Python sklearn, Python package scikit-learn, Python package Holes, Python package Scikit-learn, Python package scikit‐learn, Python package seaborn, Python package sci-kit learn, Python package scikit learn, Python package scikits.learn, Python package scipy, Python packages, Python sklearn package', 'python sklearn': 'Python Sklearn, Python sklearn, Python sklearn library, Python Sklearn library, Python3 sklearn, Python package sklearn, Python sklearn-library, Python sklearn.cluster, Python-sklearn, Python sklearn package', 'python sklearn library': 'Python Sklearn, Python sklearn, Python Scikit-Learn library, Python sklearn library, Python scikit-learn library, Python Sklearn library, Python Scikit-learn library, Python3 sklearn library, Python sklearn-library, Python ‘sklearn’ library', 'qcluster': 'ClusterR, mCluster, :Cluster, kCluster, sCluster, ClusterX, SCluster, ClusterT, ACluster, ¥Cluster, DCluster, Cluster, dCluster, ClusterW, eCluster, FCluster, Cluste, MCluster, Cluster8, cCluster, Cluster3, PCluster, hCluster, XCluster, RCluster, Clusters, xCluster, UCluster, pCluster, Cluster®, ClusterA, 1Cluster, iCluster, ClusterM, WCluster, μCluster, 5Cluster, aCluster', 'r package star': 'R package SIS, R package (R, R package STARTRAC, R package TAM, R package NST, R package GO, R package cn, R package MVST, R package STANDARICH, R package SCAN, R package VAST, R package SSN, R package SPIn, R package st, R package SIMR, R package SPARCC, R package SSDM, R package ks, R package SECR, R package DTA, R package rq, R package ARES, R package CAR, R package STATS, R package r, R package mt, R package R0, R package MAXSTAT, R package sm, R package HH, R package MR, R package SP, R package SCAM, R package MAST, R package NB, R package SSCLUST, R package Stat, R package q, R package openSTARS, R package SIBR, R package StTAMPP, R package SGL, R package ga, R package EZ, R package SOM, R package AROC, R package1, R package SEG, R package,, R package CSAR, R package BF, R package STAMPP, R package ML, R package STRING, R package VARS, R package SMUT, R package SVA, R package AF, R package BEST, R package SoDA, R package CVST, R package2, R package STACKR, R package SITAR, R package MAR1, R package SPEI, R package (, R package RAR, R package SPARSim, R package:, R package SPARTAN, R package JM, R package SAM, R package SCCA, R package FD, R package BB, R package Surv, R package ez, R package iq, R package ca, R package g, R package SLHD, R package sn, R package SIAR, R package LD, R package3, R package SICS, R package ARM, R package DART, R package SMCP, R package PK, R package SNMF, R package SAFE, R package SC3, R package SGAT, R package mi, R package STRATAG, R package BaSTA, R package BETAREG, R package BART, R package “p, R package sp, R package DT, R package SMAT, R package MI, R package SQUAREM, R package HZAR, R package FW, R package My, R package SimR, R package SC2P, R package CARE, R package lm, R package SIM, R package MST, R package SPP, R package tm, R package SEM, R package SSAR, R package sf, R package STPGA, R package CM, R package of, R package SPIA, R package np, R package CA, R package ‘STATS’, R package to, R package GA, R packaged, R package SCDE, R package FIAR, R package SKAT, R package R', 'r2dt': 'R2D', 'rainbowr': 'Rainbow, RAINBOWb, RAINBO, RAINBOW, BRAINBOW, RAINBOWc, RAINBOWRMSD, Rainbow2, RainbowRMSD', 'rdistance': 'Distances,  distance, “Distance, gDistance, Distance, Rdist, iDistance, IDistance, gdistance, distance', 'rtrack': 'Track, Rtracklayer, ARTTrack, CSRTrack, LSRtrack, RTrackLayer, track', 'scina': 'SCINA R, CINA, SCINAto', 'sklearn': 'Python sklearn, Python sklearn library, Scikit-learn, sklearn.tree, sklearn-fuse, scikits.learn, sklearn Python package, sklearn.svm, sklearn0, sklearn”, sklearn.utils, Python package sklearn, sklearn_extra, sklearn Python library, Sklearn API, sklearn Python, sklearn, sklear, Python sklearn package, sklearn.hmm, sklearning, sklearn‐rvm, sklearn python package', 'spatest': 'PATest, R package SPAtest, SPAtest R package', 'star': 'STARU, Star/2, eSTAR, StARTs, PSTAR, SSTAR, (STAR, STAR23, STAR22, Sstar, STAR2, STAR21, STAR-E, START, STARRS, STAR75, CStaR, GSTAR, STARe, QStar, stare, STAR12, Stara, ustar, Starr”, STARs, STAR-J, STAR-G, StAR2D, eStar, StA, STAR-2, STAR-T, starve, STaRS, pStAR, StarAI, star/2, CSTAR, Star-P, Qstar, STaRI, STAR-P, starpu, STAR13, STAR®, STARv, start domain-containing protein 1, STAR*D, STARRT, STARS, STAR28, iStar, StARRT, STARTT, STAR package, STAR-R, Starst, astar, STARD, star-e, STARma, STARTs, STAR), ASTAR, STAR19, Stark, STARZ, STAR51, STAREE, istar, STAR39, Starr, STAR-Q, StarFL, STAR RNA-seq aligner package, STAR:, STARP, STAR/2, rna-star, Spliced Transcripts Alignment to Reference, sta, STAR-F, STARx, Sta, STARNA, KSTAR, StARS, StaRI, StART, Stargz, STAR‐2, StarE, STAR67, DNA STAR package, STAR-C, STARTS, STARKS, STA, dStAR, STAR-D, STAAR, luteinizing hormone-induced protein, Star®, StAR2, KStar, STAR26, STARMA, STAR37, STAR3D, Spliced Transcripts Alignment to a Reference (STAR), StarS, R package STAR, StarPU, ultrafast universal RNA-seq aligner, jStar, qStar, STAR3, STARPA, JSTAR, ESTAR, StARR, STAR e, Star2D, Wstar, star, START2, StARS®, Ustar, STAR2D, Starm, STAR66, STAR©, Stare, stars, STAR*, STARD1, aSTAR, STAR_2, MSTAR, STAR58, STAR74, start2, STARFM, STARR, STAR57, StARTS, STAR +, STARch, StarT, QSTAR, STaRT, Stars, STARv2, STARC, Start, Star2, start, StarI, Starc, STARHS, STTAR, STARCH, STaRi, starka, StaRi, STAR4, STARK, STARQL, STARA, Star e, STAR5, eStAR, steroidogenic acute regulatory protein, mitochondrial, STARE, STARM, stard1, USTAR, Kstar', 'star rna-seq aligner package': 'STAR RNA‐seq aligner, STAR RNAseq aligner, STAR RNA-seq aligner, STAR RNA-Seq aligner, STAR RNA-seq aligner48, STAR (RNA-seq aligner, STAR RNA-seq Aligner', 'star package': 'STAP package, CAR package, DNASTAR package, SITAR package, DNA STAR package', 'star)': 'STAR, star, star), steroidogenic acute regulatory protein, mitochondrial, start domain-containing protein 1, luteinizing hormone-induced protein, stard1, (STAR)', 'star) rna-seq aligner': 'STAR RNAseq aligner, star, star) rna-seq aligner, STAR RNA-seq aligner, STAR) aligner, steroidogenic acute regulatory protein, mitochondrial, STAR RNA-Seq aligner, STAR RNA-Seq aligner42, start domain-containing protein 1, luteinizing hormone-induced protein, stard1, STAR RNA‐Seq aligner', 'star) aligner': 'star) aligner, STAR aligner, STAR) RNA-Seq aligner, STAR alignner, STAR aligner66, star, STAR2 aligner, STAR aligner57, STAR aligner59, steroidogenic acute regulatory protein, mitochondrial, STARaligner, STAR aligner23, start domain-containing protein 1, luteinizing hormone-induced protein, stard1, STAR aligners, STAR’ aligner', 'sentimentanalysis': 'Sentiment, SentimentAnalysis” R, Sentiment Analysis Applied, Sentiment Analysis (ABSA), Sentiment Analysis', 'seuratobject': 'Seurat Object, SeuratObjects', 'sparsedc': 'SparseM, Sparser, SparseData, SparseK, SparseDTW, Sparsey, Sparse, Spars, SparseDOSA', 'spliced transcripts alignment to reference': 'Spliced Transcripts Alignment to a Reference, STAR (Spliced Transcripts Alignment to a Reference) aligner, STAR (Spliced Transcripts Alignment to a Reference, Spliced Transcripts Alignment to a Reference (STAR) aligner, Spliced Transcripts Alignment to a Reference, or STAR, Spliced Transcripts Alignment to a Reference (STAR, Spliced Transcripts Alignment to Reference (STAR) aligner, Spliced Transcripts Alignment to a Reference (STAR) RNA-seq aligner, Spliced Transcripts Alignment to Reference (STAR-aligner), Spliced Transcripts Alignment to Reference (STARv2, Spliced Transcripts Alignment to Sscrofa11, Spliced Transcripts Alignment to Reference), Spliced Transcripts Alignment to a Reference RNA-seq, Splicing Transcripts Alignment Reference (STAR), Spliced Transcripts Alignment to a Reference (2, STAR) (Spliced Transcripts Alignment to a Reference), Spliced Transcripts Alignment to Reference (STAR), Spliced Transcripts Alignment to a Reference (STAR-2, Spliced Transcripts Alignment, Spliced Transcripts Alignment (STAR), Splice Transcripts Alignment to a Reference, Spliced Transcripts Alignment to a Reference program (STAR, Spliced Transcripts Alignment to a Reference tool (STAR, Spliced Transcripts Alignment to a Reference (STAR), Spliced Transcripts Alignment to a Reference (STAR) RNA-seq, Spliced Transcripts Alignment to a Reference) aligner, Spliced Transcripts Alignment to a Reference (STAR) RNA-Seq aligner, STAR (Spliced Transcripts Alignment to a Reference), Spliced Transcripts Alignment to a Reference (STAR) mapper, Spliced Transcripts Alignments to a Reference (STAR, Spliced Transcripts Alignment to a Reference software (STAR), Spliced Transcript Alignment to a Reference (STAR), Spliced Transcripts Alignment to Reference tool (STAR, Spliced Transcripts Alignment to', 'spliced transcripts alignment to a reference (star)': 'Spliced Transcripts Alignment to a Reference, STAR (Spliced Transcripts Alignment to a Reference, Spliced Transcripts Alignment to a Reference (STAR) aligner, Spliced Transcripts Alignment to a Reference, or STAR, Spliced Transcripts Alignment to a Reference (STAR, Spliced Transcripts Alignment to Reference (STAR) aligner, Spliced Transcripts Alignment to a Reference (STAR) RNA-seq aligner, Spliced Transcripts Alignment to Reference (STAR-aligner), Spliced Transcripts Alignment to Reference (STARv2, Spliced Transcripts Alignment to Reference), Spliced Transcripts Alignment to a Reference RNA-seq, Splicing Transcripts Alignment Reference (STAR), Spliced Transcripts Alignment to a Reference (2, STAR) (Spliced Transcripts Alignment to a Reference), Spliced Transcripts Alignment to Reference (STAR), Spliced Transcripts Alignment to a Reference (STAR-2, Spliced Transcripts Alignment, Spliced Transcripts Alignment (STAR), Splice Transcripts Alignment to a Reference, Spliced Transcripts Alignment to a Reference program (STAR, Spliced Transcripts Alignment to a Reference tool (STAR, Spliced Transcripts Alignment to a Reference (STAR) RNA-seq, Spliced Transcripts Alignment to a Reference) aligner, Spliced Transcripts Alignment to a Reference (STAR) RNA-Seq aligner, STAR (Spliced Transcripts Alignment to a Reference), Spliced Transcripts Alignment to a Reference (STAR) mapper, Spliced Transcripts Alignments to a Reference (STAR, Spliced Transcripts Alignment to a Reference software (STAR), Spliced Transcript Alignment to a Reference (STAR), Spliced Transcripts Alignment to Reference tool (STAR, Spliced Transcripts Alignment to, Spliced Transcripts Alignment to Reference', 'statistix': 'Statistica13, StatisticaPL, STATISTICA8©, Statistic®, Statistics, Statis, STATISTICA8, Statistics), Statistx, Statistix8, STATISTICA/W, statista, Statistcs, Statistiques, STATISTICA10, Statistica®7, Statistca, STATISTIKA, Statista®, Statitix, StatisticXL, Statistix10, StatistiXl, Statistica°, STATISTICA7, STATISTIX8, statis, Statistixl, Statistica8, statistix, statistiXL, STATISTICS®, StatistXL, Statistics:, Statistica X, Statistica6, Statistic \"R, STATISTICs, StatistiKL, STATISTICA15, STATISTICA6, StatisticaTM, Statista, Statistika, Statistics,, STATISTICAL, STATISTICA7©, Statisti, Statistica®8, STATISTICA®7, STATISTCA, Statistica, Statistico, Statista3, Statistika®, Statistica12, Statisticav, StatistiXL®, Statistic-z, StatisticTM, Statistica/W, STATISTICATM, StatistX, statistal, Statistica11, Statisticon, STATISTICA13, Statisti×, Statistica9, STATISTICA, STATISTIK, STATISTICS, Statistix-10, Stati, STATISTA, statistic, StatistiXL, Statistica1, statistical, Statistique, StatisticaR, Statistix7, Statistix, StatisticaPl, Statistica7, Statistix®8, Statistica-7, STATISTIC, STATIX, Statistic, Statistics®, Statistica16, statistica°, Statistica©, Statistica5, Statistical, StatisticW0, Statistix-8, statistica, STATISTICA®, Statistix®, Statistician, Statistix-9, Statisticа, STATISTICA12, STATIST, STATISTICA-6, STATISTICA©, STATIS, Statistica10, Statist, Statistix9, Statistica®, Statistica-8, statistix9, statistics, STATISTICA9', 'treetoreads': '', 'tunepareto': 'TunePareto R package', 'velvet': 'velvet, Velvet50, velvet-1, VelvetSC, Velvet58, aVelvet, D-Velvet, Velvet-CAP3, Velvet’, velvet1, VelvetSC60, Velvet32, Velvetleaf, VelvetOases, VELVETG, velveth, Velvet26, Velvet-Cap3, Velvet30, Velvet_1, Velvet®, Velvet75, Velvet-S, Velvetver, Velvetbean, Velvet29, VELVETH, Velvet21, Velvetopt, Velvet_v1, Velvetk, velvet-sc, Velvet-sc, Velvetg, Velvet42, Velvetde, VelvetK, Velvet24, Velveth, Velvet34, velvetg, Velvet38, Velvet-SC, Velvet de, Velvet81, Velvet-1, Velvet1, Velvet +, Velvet23, Velvet43', 'weightedroc': 'Weight, Weighted', 'activity': 'Activity Base, ‘activity’, ActivityScore, Activity Recall, Activia, Activity Log, Activio, Activ, ActivityBase, ActivityLoop, ActivityInfo, Activity List, activitybase, ActivityX16, Activities, ActivityCoach, Activity Survey, ActivityTracker, ACTIVITY [UNK], Activity Wheel, Inactivity), Activity Linker, ACTIV, Activitygram, ActivityAware, MyActivity, Activity Theory, Activity System, activity-atlas, ActivityProbe, ActivityStat, ActivityScope, My Activity, Tractivity®, Activity-Sim, activity cage, Activity2Vec, ACTIVITIES, BioActivity, ActivityNet, “activity, Activity, OF Activity, ACTIVITYGRAM, CReActivity, Activitybot, Activity Score, ACTIVITYBASE, Activity Atlas, ActivityIndex, Activity3D, Activity (RSA), ActivityMeasure, Tractivity, ActivityBase XE, Activity Map, Activity Plot, ACTIVIA, Activiteit, Activity Plots, ACTIVITAL, Activiti, Activité, activity, ActivityMonitor, Activity base, ActivityManager, ActivityResNet, PROACTIVITY, Activity Coach, Activity Scan, activity-stat', 'ads': 'AdSV, aDSM, ADSA, ADSM, ADSO, ADSP, ADSV, aDST, AdSS, ADST, ADSC, AdSC, ADSS, ADS)', 'ampir': 'Vampir', 'anocva': 'anocva, rmANOCVA, ANOCVA R, Analysis of Cluster Variability, ANalysis Of Cluster VAriability', 'assignpop': 'assign, Assign, assignPOP R, assignPOP R package, assignPOP, R package assignPOP, AssignPOP, assignPOPwere, AssignPop', 'beautifulsoup4': 'BeautifulSoup, Beautiful Soup”, beautiful-soup, BeautifulSoup9, Beautiful, beautifulsoup, BeatifulSoup, Beautiful Soup, Beautiful soup', 'beepr': '', 'biopy': 'biopython', 'biosignalemg': 'biosignalEMG package', 'bootlr': 'boot', 'cregulome': 'Regulomes, scRegulome, Regulome, cRegulome R package', 'catmap': 'Catman, CATM, Catma, CATMAT, CATMAN, CatM, CATMA, Catmap.pl', 'cellranger': 'cellranger aggr, cellranger, CellRangerv3, Cell Ranger, CellRanger mkfastq, CellRanger-atac, Cellranger suites, cellranger mkfastq1, CellRanger ATAC, CellRanger mm10, Cellranger, CellRanger1, cellranger count, CellRanger VDJ, Cellranger aggr, cellRanger-atac, Cell-Ranger, 10×Cellranger, CellRanger pipeline, cellranger mkfastq, Cellranger VDJ, cellranger R-kit, cellRanger mat2csv, CellRank, cellranger vdj, Cellranger ATAC, CellRanger, CellRanger aggr, cellranger-arc, Cellranger pipeline, CellRanger Pipeline, cellrangerRkit, CellRanger2, Cellranger‐atac, cellrangerRrkit, Cell Rangers, cellranger suite, Cellranger V(D)J, Cellranger mkref, cellRanger, CellRanger R, cellRanger aggr, Cellranger (10X, CELL RANGER, Cellranger mkfastq, Cell Ranger3, cell ranger, Cell ranger, Cellranger-atac, CellRanger mkref, cellranger-atac, Cellranger count, cellranger mkref', 'cgraph': 'graph, ECGraph, Graph', 'chemometrics': 'Chemometric, chemometrics, Chemometrics, R-chemometric, ChemometricsWithR, chemometric, “chemometrics', 'cobs': 'R-package cobs, cob, COOBS, COB', 'cumseg': '', 'cytoflow': 'Cytof, Cytoflow, cytoflow, Cytofworkflow, cytofWorkflow, cytof', 'dae': 'DAEI, DAEO, DAEL, DAEA, DAEM', 'data.table': 'data.table_1', 'deepredeff': 'R package deepredeff', 'django-blastplus': '', 'dplyr': \"R-package ‘dplyr, dplyr package, R package “dplyr”, dplyr R package, R package ‘dplyr’, dplyr19, dplyr R, R package “dplyr, dplyr, dplyr_0, R package dplyr, R packages ‘dplyr’, R Package dplyr, PLYR, R-package dplyr, R library “dplyr, dply, dplyr’, package dplyr, R package 'corrr' and 'dplyr, plyr, dplyr R-package, dplyr42, R library dplyr, R package ‘dplyr, R packages dplyr\", 'eqtlseq': '', 'earth': 'Earthrise, Earthstat, Earthmate, EarthEnv, Earthworm, Eartha, Earthstar, EarthData, earth Pro, EarthStat, EarthCARE, EART, Earthv1, Earthlink, Earthdata, EarthChem, EarthCape, R-package “earth, EarthSat, earthpy, Earthox, Earthmine, EARTHDATA, EARTHTIME, MEarth, EarthZyme, Earthscan, EarthTech, Earth-WS, EARTHSTAT, EarthCam, Earth®, EarthDNA, Earthtm, Earth Pro, EarthCare, EarthLook, earth R, EarthCube, Earthsky, EarthTec®, Earth3P, Earther', 'edd-utils': '', 'enetlts': 'R package enetLTS', 'fastdummies': '', 'fullfact': 'fullfact package, Full Fact', 'funfem': '', 'furniture': '', 'fwdpy': 'fwdpy11', 'gdscad': '', 'geobr': '', 'ggnewscale': '', 'ggplot2': 'R package \\'ggplot2\\', R ggplot2 environment, R graphics package ggplot2, R package {ggplot2, R package for visual analytics (ggplot2, ggplot227, ggplot2 R graphics, ggplot268, ggplot2 R library, ggplot266, R-package ‘ggplot2′, ggplot264, ggplot2t, ggplot289, ggplot218, ggplot255, R ggplot2 package, ggplot271, R project ggplot2, GGPlot, ggpubr (ggplot2 Based Publication Ready Plots) of R, R package ‘ ggplot2, ggpplot, R package ‘ggplot2′, Ggplot, r package ‘ggplot2, R library ggplot2, CRAN R package ggplot2, R package ggplot2, R package of ggplot2, R-ggplot2, R package \"ggplot2\", R package (ggplot2, R/Bioconductor package ggplot2, R-package ggplot2 -, Ggplots, ggplot256, ggplot226, ggpllot2, ggplot241, ggplot2-v3, R package ‘ggplot2’, ggplot2for, ggplotfy, R ggPlot2, ggplot2bdc, R package “ ggplot2, R/ggplot2, ggpplot2, ggplott2, R package “ggplot2”, ggplot231, R graphing package ggplot2, ggplot2180, ggplot244, ggplot261, r package “ggplot2, R “ggplot2, ggplot262, ‘ggplot2, ggplot2 package of R, R package: ggplot2, ggplots2, R package ggplot2 package, R-package “ggplot2, ggplot5, R function ggplot2, ggplot R, ggplot2, ggPlot, ggplot240, ggplot2R, GGplot2v1, ggplot2_2, R / ggplot2, ggplot221, R Packages ggplot2, ggplot246, RStudio package ggplot2, R “ggplot2 package, ggplot26, ggplot.2, ggplot2 R-package, R ‘ggplot2′, ggplot254, ggplot2 R language, R ‘ggplot2, ggplot234, ggplot2 package, R-package ‘ggplot2’, ggplot2 package x, R package “ggplot2, PCA ggplot2 R, ggplot2_3, Ggplot2®, ggplot1, ggplot22, ggplot243, ggplot215, ggplotly, ggplot (ggplot2 R package, ggplots, ggplot251, R package \"ggplot2, R ggplot2, ggplot2.R, ggplot2”, R Cran packages ggplot2, GGplot, ggplot2′ R, R studio ggplot2, ggplot2” R package, ggplot2 package R, “ggplot2, ggplot21, gplot2, R-package “ggplot2”, R plotting system ggplot2, ggplot258, ggplot248, R Package ggplot2, ggplot, ggplot2 R package, ggplot2 (R, R language ggplot2, ggplot2′, R-package ‘ggplot2, ggplot2s, KMggplot2, ggplot237, R “ggplot2”, R-package ggplot2, package “ggplot2, R packages ggplot2, package ggplot2, GGPLOT, ggplot219, ggplot2; package, ggplot2 r, ggplot2ʹ, R package ‘ggplot2, ggplot2485556, \\'ggplot2, Ggplot2 R, ggplot2 R, ggplot2 package geom_points, ggplot252, ggplot245, R library ‘ggplot2, ggplot2 package for R', 'ggpmisc': 'ggpmisc package', 'ggsci': 'R package ggsci', 'gitlab': 'GitLab-CI, gitlab, GitLab CI', 'hab': 'HABE', 'httplib2': 'httplib', 'insol': 'Insole, Insolvenz, Insole-C, Insoles', 'intepareto': 'R package intePareto', 'interep': 'interest, inter', 'intsvy': 'R package intsvy', 'knitr': 'knitr package, Knitr R, R package “knitr”, Knitro, knitr R, knitr R package, R knitr, knitr1, R package knitr, knitr R library, R package “knitr, knitr, knitr Rnw, R knitr package', 'kontroller': 'controller, Controller', 'libmesh': 'libMesh41, libMesh40', 'logspline': 'logspline R package, R package “logspline”', 'lubridate': 'lubridate R, lubridate, lubridate R package, Lubridate, R package lubridate', 'lunar': 'Lunar®, LUNA, LUNARISTM, Lunarc, LUNAR DPX, Luna, Lunar DPX', 'margins': 'R package margins, MARGINAL, MARGIN, Marginal, MARGINSPLOT, Margin, MARGINto, R package “margins, MARGI', 'matplotlib': 'MATPLOTLIB34, matplot-lib, MatPlotLIB, Matplotlib/Seaborn, matplotlib-venn, Matplotlib (MPL), matplotlib69, matplotlib Python package, matplotlib53, Matplotlib41, Matplotlib ScaleBar, Matplotlib basemap, Matplotlib-venn, Matplotlib3, MatPlotlib, python plotting, Matplotlib36, matplotlib Python, matplotlib package, Matplotlib package, matplotlib42, matplotlib Axes, matplotlib, Python matplotlib library, Matplotlib2, matplotlib1, Matplotlib5152, matplolib, Matplotlib.pyplot, Matplolib, Matplotlib-v3, MatPlotLib graphics, Matplotlib.mlab, Matplotlib1, MatplotlibLicense, matplotlip, Matplotlib56, MatPlotLib, mathplotlib, Matplotlib26, Matplotlab, Matplotlib Basemap, MathPlotLib, Matplotlib BaseMap, Matplotlib Pyplot, matplotlib.mlab, Matplotlib-1, Matplotlib33, Matplot, matplotlib Basemap, matplot, matplotlib plotting, Matplotlib, matplotlib Python library, Matplotlib©, Matplotlib Library, Matplotlib63, Matplotlib-package, Matplotlib MEC App, matplotlib 2D, matplotlib-mlab, MatPlotLib40, Python matplotlib, Matplotlib library, matplotlib hist2d, matplotlib.pyplot, Matplotlib Python, Python matplotlib package, matplotlib_venn, Matplotlotlib, Matplotlib pyplot, Matplotlib_venn, MatplotLib, Matplotlib Toolkit, matplotlib_ven, matplotlib45, matplotllib', 'methylcheck': 'methyl, MethylC, Methyl', 'mirt': 'miR, miRTop, R package “mirt, MIRTLE, MIR, miRTif, miRTDL, mirTAR, MIRTCR, R packages mirt, miRTRS, R mirt package, miRTS, R package mirt, mirt R-package, R-package ‘mirt’, mirt R, Mirth, Mir, miRTOP, BMIRT, R-package mirt, mirTex, MIRT R, mirTar, miRTAR, R library mirt, miRTil, miRTar, miRTex, mirt package, MIRTA, miRTIF, mirt R package, miRTaH, MIRTE, miRTI, R mirt, MIRTEX, Mirtar, “mirt, mirT2, R-package “mirt, MIRTK', 'mixkernel': 'mixKernels', 'mnnpy': '', 'modes': 'Modeshape, modest, kmodes, modeest, gmodes, Modest, MODE, ModestR®, ModestR, ODES, MODESLAB, Modetest, Mode, MODESTR®, MODESE, ModesLab, mode, MODEST, MODESA', 'mokken': 'R package ‘mokken’, R “mokken”, Mokken’, “mokken, R package “mokken”, “Mokken”, mokken package, R-package ‘mokken’, R package mokken, mokken”, R mokken package', 'mord': 'morda, mor', 'mpi4py': 'MPI4PY, mpi4py Python, MPI4Py', 'mrmlm': '“mrMLM, R package mrMLM, rMLM, R package “mrMLM”, mrMLM R, R package “mrMLM, MRML, RMLM, mrMLM R package, mrMLM.GUI', 'ndl': 'NDLA, NDLP', 'ngcgh': 'ngCGH python', 'nlsic': 'nlsi, LSIC', 'nolds': 'Nold', 'numpy': 'numpy.fft, Numpy≥1, NumPy8, NumPy4344, numpyc, numpy1, Numpy3, numpy®, numpy67, NumP, Numpy API, numpy python package, numpy Python package, Python numpy, NumPy/C, NUMP, NumPy for, numpy-1, numpy39, numpy15, NumPy3, NumPy V, numpy, NumPy1, numpy Python library, numpydoc, Numpy©, numpy python, Gnumpy, NumPyDoc, NumPy-1, Numpy1', 'onnx': 'ONN', 'openrouteservice': 'Open Route Services, OnRouteService, Openrouteservice, openrouteservice', 'optrees': '', 'osmdata': 'R package “osmdata”, R package osmdata', 'paramtest': 'R package paramtest', 'patchwork': 'PATCH, PatchworkCG, patch, Patchwork ASCNA, Patch, Patchwork.plot, patchwork, Patchwork', 'pdftables': '', 'phylip': 'PHYLIN, phyl, phylips, phylipFasta, phylip.tree, Phylip-69, Phylips, phylip_var, PHYLIP3, PHYLIP-3, PHYLIP80, hylip, phylip), (PHYLIP), PHYLIP suit, phylogeny inference package, Phylipand, PHYLIP24, PHYLIP55, phylipout, PHYLIP60, Phylip), Phylip4, Phyllip, PHYLIP (ML), Phylip+NJ, PHYLIP pars, Phylip2Tree, Phylip, PHYL, phylip, PHYLogeny Inference Package, phylin, Phylip NJ, Phylip Tree, PHYLIPNEW, PHYLIP3.65, PhyliP, PHYLIB, PHYLIPv3, PHYLIS, phylip MSA, Phylip Ver, Phylip56, PHYLIPS, rphylip, Phylip-3, Phylip Plot, phylip -T, Phyl, 2PHYLIP, PHYLIP71, PHYLIP-ML, PHYLIP52, Rphylip, phylip-3, PHYLIP40, Phylip3, RPhylip, phylip3.6, Phylip_REF, phylip -m, Phylip‐3, PHYLIP51, Phylip-v3, phylip’, PHYLIP)', 'powergwasinteraction': 'R package powerGWASinteraction', 'psych': 'R-package psych, R psych, Psychlit, psychmeta, R library psych, package psych, PsychNet, PsychoAge, R ‘psych, psych R library, iPsych, Psychnet, R psych package, R “psych, PsychoPy1, R language psych, psycho, R packages psych, psych Package in R, PSYC, PsychSocs, NPSYCH, psych R, PsychCore, PsychoPy2, psychenet, Psych*, Psyche, PsychoPy3, Psychopy2, psych R-package, Package “psych” - R, PSYCHLOP, Psycho, psychlit, PsychData, psych r-package, R package (psych, PsychoPy, RStudio psych package, “psych, R package ‘psych, R psych package36, PsychTr, R package \"psych\", Psych Soc, Package psych, PsychE, PsychoFit, PsychRNN, PSYCHIC, PsychPy, PsychINFo, PsychINfo, R package: psych, psychache, Psych-DS, Psychpy, Psychdata, Psyc, Psych MD, psychopy, PsychBITE, psyche, PsychTest, psych’, Psychmeta, PsychLog, PsychLab, PSYCHinfo, Psychosis, PsychGene, psych’ R package, Psychlab, R psych library, PsychInfo, PsychLIT, psychInfo, PsychSIGN, Psych R, Psychinfo, Psychache, r-package psych, PsychoJS, iPSYCH, Psychotic, PsychHID, R-package “psych, psych package for R, R ‘psych’, PsychMap, Psychopy3, PSYCHO, PsychSim, r package psych, R library ‘psych’, psych, psych r, PsychCorp, Psychorus, PsychMAP, R Software Packages psych, PSYCHE, R package ‘psych’, PsychChip, psychoEQ, Psychopy®, PsychNET, PsychLit, Psychlops, PsychicVR, Psychid, PsychPark, PSYCHLOPS, psych package, R “psych”, R package psych, PSYCHLOPs, Psych Map, Psychopy, Psych Net, R-package “psych”, PSYCHInfo, PsychSci, Psych-soc, R project psych package, R package “psych, R ‘psych’ package, psych”, Psychenet, psych Package, R-package ‘psych’, PsychCare, PsychoSig, PSYCHLAB, psych R package, R package “psych”, R package, psych, PsychINFO, PsychoTSH', 'pvr': 'R package ‘PVR’, PVRI, R package “PVR, R-package PVR', 'pyafq': '', 'pymagnitude': 'magnitude', 'pyphysio': 'Physio, pyphy, pyphysio, Pyphysio, PyPhy, Pyphy', 'pyqi': '', 'pytorch': 'PytorchVideo, PyTorch C, Pytorch-GPU, PyTorch DL, Pytorch-0, PyTorchVideo, pytorch, PyTorch-Kaldi, PyTorch1, PyTorch GPU, Pytorch0, PyTorch3D, pytorch-1, PyTorch5, Pytorch Adam, Pytorch3D, PyTorch2, PyTorch-1, pytorch1, PyTorch0, PyTorch-NLP, Pytorch1, GPyTorch', 'pytplot': '', 'rbayesianoptimization': 'rBayesianOptimization, R package “rBayesianOptimization”, rBayesianoptimization, BayesianOptimization', 'rclinicalcodes': 'rClinicalCodes R, ClinicalCodes', 'rlidar': 'LiDAR', 'readr': 'readr package, readr R package, read, R-package readr, readrr, Read, R packages readr, R package readr', 'relaimpo': \"relaimpo R, relaimpo R package, R “relaimpo, R package ‘relaimpo’, relaimp, R package relaimpo, R package “relaimpo”, R package ‘relaimpo, relaimpo [R], R package 'relaimpo', Relaimpo, R package “relaimpo, relaimpo package, R-package relaimpo, R library relaimpo, relaimpo, r package relaimpo\", 'requests': 'CRequest, requests, Requests, grequests, python requests library, Request, RequestWS, Request a', 'rglobi': '', 'rmarkdown': 'RMark, rMarkdown, RMarkdown, R/RMarkdown, RMarkDown, Markdown, rmarkdown, R package rmarkdown, Rmark, Rmarkdown, markdown, MarkDown, R-package rmarkdown', 'sccatch': 'R package scCATCH', 'scikit-learn': 'Scitkit-Learn, SciKit-learn, scikit-learn Python, scikit learn, scikit-learn python library, scikit-learn Python Library, Scikitlearn, Scikit-learn (Machine, Scikit‐Learn, scikit-learn: machine learning in python, scikit-learn library, Scikit-Learn®, scikit-learn Pedregosa, scikit-learn estimator, Scikit-learn:, scikit-learning, Python scikit-learn, scikit-learn library for Python, scikits-learn, scitkit-learn, scikit-learn Python library package, scikit -learn, Python scikit-learn gridsearchCV, Scikit-learn Pedregosa, Scikit learn, scikit-learn10, scikit-learn, SciKit-Learn, Scikit-Learn, Scikit-Iearn, Scikit-Learn/manifold, Python library scikit-learn, Scikit_learn, Scikit-learn package, Scikit-Learn.PCA, Scikit-learning, Scikit-learn, scikit‐learn, scikit_learn, Scikit-learn Package, Python package scikit-learn, Scikit-learn”, Scikit-Learn library, Scikit-Learn.FastICA, Scitkit-learn, scikit-learn Python package2223, Scikit–learn, ScikitLearn, Scikit-learn Python, Scilit-learn, Scikit-Learn Python, scikit-learn81, Scikit‐learn, 2scikit-learn, Scikit-Learn Package, scikit-lear, Scikit-learns, Scikit-learnv, scikit-learn Python package, scikit-learn Developers, Scikit-learn Toolbox, sci-kit-learn, scikit-learn Python package for, Sciket-Learn, Scikit-learn29, scikit-learn python, SciKitLearn, Scikit-Scikit-learn, Scikit-learn0, sciki-learn, Scikit Learn, SciKit Learn, scikit-Iearn, Scikit-learn library, Scikit-Learn Library, SciKit-Learn library, Scikits-learn, scikit-learn2, scikitslearn, scikitlearn, scikit-learn CART, SCiKit-learn, Scikit-learn Library, scikit-learn22, Scikit-Learn package, Python scikit-learn package, scikit-learn (Pedregosa, Scikit-lean, SciKit learn, Scikit-Learn PCA, scikit-multilearn, scikit-learn package, Scikit-learn python, Scikit-learn toolbox, scikit-learn-contrib, Scikits-Learn, Sciket-learn, sckit-learn, scikit-learn33, scikit-learn python package, scikit.learn, scikit-learn: machine learning in Python, Scikit-learn31, Python scikit-learn library, Sckit-learn, scikit-learn48, Scikit–Learn, Scikit-Learn24, SCikit-learn, Sci-kit-learn, scikit-learn R package, scikit-learn (sklearn), Scikit-learn (Pedregosa, scikit-learn8, scikit-learn API, scikit - learn, Scikt-learn, Scikit Learn5, scikit-learn Python library, scikit-Learn', 'sets': 'hla-dr-associated protein ii, template-activating factor i, taf-i, phosphatase 2a inhibitor i2pp2a, phapii, igaad, i-2pp2a, protein set, set, esets, SETSM, sets, SET, inhibitor of granzyme a-activated dnase, Set', 'sfsmisc': \"Package sfsmisc, R package 'sfsmisc', sfsmisc R-package, sfsmisc package, R package sfsmisc, sfsmisc R, R package “sfsmisc, R/sfsmisc\", 'shiny': 'shinyngs, R-package shiny, shinyGPA, ShinyGEO, shiny, Shinyjs, glossy, Shiny-seq, shinyTree, ShinyGeo, ShinyGPA, SHINY R, shinyHTM, shinyBN, shinylogs, shinyDAG, shiny R, ShinyAIM, Shiny app, Rshiny, ShinyGo, Shiny/R, ShinyStan, ShinySOM, shinysky, SHINY app, shinyFuse, shinyMA, ShinyNGS, rshiny, shinyLi, SHINNY, Shiny-app, ShinyApp, shinyLi R, ShinyGO®, shinyAce, ShinyNANO, shiny18, RShiny, ShinyRGL, shiny.io, Shiny3, glistening, Shiny-R, Shiny App, Shiny GAM, shinyjs, Shinymeta, shinyApp, Shiny-Seq, ShinyGO, ShinyIO, Shiny.r, ShinnyGO, ShinyJS, ShinyR, shinyAE, Shiny-GO, R-package ‘shiny’, ShinyApps, shinymeta, Shiny27, Shiny R, shinyGEO, ShinyLi, Shiny Web, ShinyGOv0, ShinyGPAS, SHIN, ShinyBN, shinystan, shinyBS, Shinystan, Shiny GUI', 'sklearn python': 'sklearn Python, Sklearn Python, sklearn Python package, sklearn.utils Python, sklearn Python library, Sklearn of Python', 'sklearn python package': 'hmmlearn Python package, learn Python package37, Nilearn Python package, sklearn Python, sklearn Python package, sklearn python package, sklearn package', 'smnet': 'MNet, PSMNet, MNET', 'snpready': 'snpReady package, r package snpReady, snpReady R, snpReady R package, SnpReady, R package snpReady, R package “snpReady, snpReady, SnpReady R', 'spbayessurv': 'R package spBayesSurv, spBayesSurv, spBayessurv, spBayes', 'spacesxyz': 'space', 'spektral': 'Spektr, spektr, Spektraris, Spektral, spektral', 'sst': 'SSTA, SSTM, SSTD, sstr, SST3, SSTS, SST9, sstl, SSTY, SSTW, SSTN, sst2, SSTF, SST+, SSTp, SSTT, SSTR, SSTP, SSTC, SSTO', 'stars': 'StarSEQ, Stars4ALL, stars, StarSolo, start domain-containing protein 1, STARRseq, starsolo, STARSIL®, Stars4All, StarSeq, STARSurg, STAR, STARRS, Star, StarScan, STARS4ALL, luteinizing hormone-induced protein, (STARS, StarStuff, STARSIM, Starst, StarStim8, STARshort, StarStim, Starstim, Starseq, TARS, StarSeach, StarStim®, STARSHIP, Starship, StarSpire, Starstim®, STARscan, star, StAR, StARS®, steroidogenic acute regulatory protein, mitochondrial, STaR, Starseq®, Starsoft, STARsolo, StarSpace, stard1, STARSEQ, STARSolo, STARSTIM', 'stdreg': 'R-package stdReg', 'strand': 'strandBias, StrandOmics, strandCet R, StrandNGS, STRAND7, Stra, H-Strand, Strand-Rx, Strand7®, Strand –NGS, STRAND®, Strand-Seq, Strand7 FEA, Strand-NGS, Strandseq, Strand NGS, STRand46, Strans, STRANG, Stranded, strand NGS, Strands, StrandOMS, STRAND NGS, Strand7, STRA, Strand-seq, STRAND-Seq, Strand LS, Strand NGS®, STRAND® NGS, strandCet, STRANDS, Strandomics, StrandNet', 'stringr': 'String10, stringR, stringer, StringGO, Strings, stringr R package, string11, stringr, Strin, String*, Stringdb, stringdb, StringDB, stringDB, stringi, R package stringr, StringDb, Stringi, string, stringr19, String©, String”, String, String-d', 'summarytools': 'Summary, Summarytools, summary, Summary Tools, summarytools', 'svunit': '', 'swirl': 'swirlify, SWIR', 'swirlify': 'swirl', 'tandem': 'Tandem-L, x!Tandem, Xtandem, XTandem, Tandem2, Etandem, etandem, ! Tandem, TANDEM-IRI, Tandemgenes, X!Tandem, inTandem, TanDEM IDEM, XTANDEM, TAND, TandemSmart, TANDEM51, Xtandem!, X1Tandem, TandemHeart, !TANDEM, ETANDEM, Tandem-R, rtandem, XTandem!, TAnDem.Plus, TanDEM-L, Tandem Play, TandemDupli, TandemQUAST, TandemQuery, TandemSWAN, Tandem-X, tandemMOAC, TanDEM-X, Tand, XITandem, TandemTools, tandemquast, eTandem, xtandem, TandemUTR, Tandem-HTRF, X!tandem, x!tandem, MCtandem, X!TANDEM, X-Tandem, ANDEM, Tandem!, E-tandem, Tandem2XML, rTANDEM', 'tensorflow': 'Tensorflow-hub, TesnsorFlow, Tensorflow Python, TensorFlow25, Tensorflow-Lite, Tensorflow Serving, TensforFlow, Tensorflow®, Ten-sorflow, TensorFlow lite, TensorFlow keras, Tensor-flow, Tensorflow Library, TensorFlowSharp, Tensorflow-GPU, TensorFlow Fold, TensorFlow/Keras, tensorflow 2, Tensorfow, Tensor flow, tensorFlow, TensorFlow2, TensorFlow-GTP, Tensorflow Deeplab, TensorFlow NN, Tensorflow-addons, RSTensorFlow, TensorFlow Mobile, TensorFlow library, TensorFlow Lite for, tensowflow, Tensorflow-0, Tensorflow Demo, tensorflowsharp, tensorflow Python, TensorFlow-package, Tensorflow Hub, TensorFlow Eager, Tensorflow Visor, Tensorflow50, TensorFlow-Keras, tensorflow1, tensorflow-gpu-2, Tensorflow ( Abadi, Tensorflow GPU, TENSOR, TensorFlow Serving, TensorFlow_gpu1, tensorflow GPU, TensorFlow13, Tensorflow2, Tensorflow-gpu1, TensorFlow/Theano, TensorflowLite, TensorFloat, TensorFlow SDK, TensorFlow11, tensorflow-gpu, TensorFlow DeepSurv, Tensorflow1, TensorFlow train, Tensorflow/Keras, TensorFlow Lite, Tensorflow.js, tensorflow library, TensorFlow1, TensorFlow.js, TensorflowJS, TensorFlow-GPU, TensorFlow15, TensorFlowPrivacy, TensoFlow, TensorFlow (TF), TensorFlow Hub, TensorflowHub, Tensorflows, TensorFlowTM, tensorflow2, Tensorflow Lite, Tensorflow2/Keras, TensorFlow DL, TensorFlow38, Tensorflow-Keras, TensorFlow/Python, Ten-sorFlow, Tensorflow-r0, Tensorflow library, Tensorflow API, TensorFLow, tensorflow API, TensorFlow Python, Tensorflow-gpu, Tensor, TensorFlow-lite, Tensor Flow, TensorFlow Core, Tensorflow Graphics, tensorflow, TensorFlow API, tensorflowTM, tensorflowlight, TensorFlow‐Gpu, Tensor-Flow, TensorFlow JS, tensor flow, TensorFlow-Slim, Tensorflow Mobile, TensorFlow GPU, TensorFlow Light, Tensorflow, tensorflow (keras), Tensorflow ADADELTA, Tensorflow lite, Tensorflow_GPU, TensorFlow toolbox, Tensorflow Keras, Tensorflow-Hub, TensorFlow package, TensorFlow, TensorFlow Keras, rTensorflow, TensorFlowLite, TensorFlow-gpu', 'tidyr': 'Tidy, tidyr R package, tidyr43, tidyr41, tidyr, package tidyr, tidyr”, tidy, R packages tidyr', 'tidyverse': 'R tidyverse package, R packages tidyverse, R package “tidyverse, Tidyverse package, R package tidyverse, R package ‘tidyverse, tidyverse package, tidy-verse, tidyverse package bundle, tidyverselibrary, R Tidyverse, R ‘ tidyverse’, tidyverse36, R ‘tidyverse, tidyverse R, R/tidyverse, R package “tidyverse”, tidyverse bundle, tidy verse, tidyverse”, Tidyverse, Tidyverse image, R package ‘tidyverse’, R ‘tidyverse’ package, tidyverse-1, Tidyverse R, R tidyverse, tidyverse, tidyverse library, package tidyverse, R library tidyverse, tidyverse R package', 'tigerstats': '', 'treehouse': 'Treehouse OCTAD, Treehouse, Treehouse Maps, Treehouse CARE, treehouse', 'tripal': 'trip, TripalDock, Trip, TripalMap, TripalField', 'truncreg': '', 'tsutils': '', 'venn': 'jvenn, VENNY2, venny2, Venny, JVENN, eVenn, VEN, RVenn, EVenn, Venny2, Venn2, venn3, pvenn, Vennt, VENNY, venn R, Venncy, Venni, venny, jVenn, DVenn, Jvenn, Ven, Vennix, PVENN, VENNy, JVenn, 3Venn, nVenn', 'whoa': ''}\n"
     ]
    }
   ],
   "source": [
    "# Add synonyms column\n",
    "benchmark_dictonary= get_synonyms(benchmark_dictonary,1,1)\n",
    "print(benchmark_dictonary)\n",
    "benchmark_df[\"synonyms\"] = benchmark_df[\"name\"].str.lower().map(benchmark_dictonary)\n",
    "\n",
    "# Save the enriched benchmark\n",
    "benchmark_df.to_csv(\"../temp/v3.1/benchmark_with_synonyms.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
